compiler/rustc_middle/src/ty/consts.rs

```diff
@@ -311,7 +307,7 @@ impl<'tcx> Const<'tcx> {
         tcx: TyCtxt<'tcx>,
         param_env: ParamEnv<'tcx>,
         span: Span,
-    ) -> Result<ValTree<'tcx>, ErrorHandled> {
+    ) -> Result<(Ty<'tcx>, ValTree<'tcx>), ErrorHandled> {
```

Comment:
```suggestion
    ) -> Result<(ValTree<'tcx>, Ty<'tcx>)>, ErrorHandled>
```

or tbh, maybe add

```rust
struct ConstValue {
    data: ValTree<'tcx>,
    ty: Ty<'tcx>,
}
```
would be open to leave that for a separate PR though, we should merge this fairly quickly as it's quite prone to merge conflicts

Reply:
nods will follow up with this